discovery-infra/utils.py

```diff
@@ -113,6 +113,16 @@ def are_all_libvirt_nodes_in_cluster_hosts(client, cluster_id, network_name):
     )
 
 
+def are_libvirt_nodes_in_cluster_hosts(client, cluster_id, num_nodes):
```

Comment:
u don't need to check that those are hosts that u expect them to be there?

Reply:
it can be problematic, since i don't know how to differentiate the worker nodes: some of them might be in installed cluster and not in the day2. i think that if we wanted to boot n workers and we see n workers registered in the cluster - then it is a good enough check for me